### PR TITLE
IDEX-2666: Fixed run extension on Windows OS

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**

### CODE REVIEW

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/WindowsTomcatServer.java [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        try {
+            generateSetEnvScript(appDir, runnerCfg);
+            startUpScriptFile = generateStartUpScript(appDir, runnerCfg);
+            Files.createDirectory(logsDir.toPath());
+        } catch (IOException e) {
+            throw new RunnerException(e);
+        }
+        final List<java.io.File> logFiles = new ArrayList<>(1);
+        logFiles.add(new java.io.File(logsDir, LOG_FILE));
+        return new TomcatProcess(appDir, startUpScriptFile, logFiles, runnerCfg, callback, codeServerProcess, eventService);
```
Comment (skabashnyuk) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/WindowsTomcatServer.java]: ImmutableList.of(new java.io.File(logsDir, LOG_FILE))

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServer.java [hunk 2]
```diff
@@ -51,307 +19,11 @@
...
-                        } catch (IOException e) {
-                            LOG.error("Unable to update mirror of project {} for GWT code server.", event.getProject());
-                        }
-                    }
-                });
-            }
-        }
-    }
+public interface CodeServer {
+    CodeServerProcess prepare(Path workDirPath,
```
Comment (skabashnyuk) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServer.java]: javadoc

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerImpl.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * GWT code server abstract implementation.
+ *
+ * @author Artem Zatsarynnyy
+ */
+public abstract class CodeServerImpl implements CodeServer {
```
Comment (skabashnyuk) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerImpl.java]: AbstractCodeServer

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerImpl.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            final java.io.File file = Utils.exportProject(projectDescriptor, extensionSourcesPath.toFile());
+            ZipUtils.unzip(file, extensionSourcesPath.toFile());
+
+            return start(workDirPath.toFile(), runnerConfiguration, extensionSourcesPath, projectDescriptor.getBaseUrl(), executor);
+        } catch (IOException e) {
+            throw new RunnerException(e);
+        }
+    }
+
+    protected abstract CodeServerProcess start(File codeServerWorkDir,
```
Comment (skabashnyuk) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerImpl.java]: javadoc

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerProcess.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        this.projectApiBaseUrl = projectApiBaseUrl;
+        this.executor = executor;
+    }
+
+    public abstract void start() throws RunnerException ;
+
+    public abstract void stop() throws RunnerException ;
+
+    public void getLogs(Appendable output) throws IOException, RunnerException {
+        final String url = bindAddress + ':' + port + "/log/_app";
```
Comment (skabashnyuk) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerProcess.java]: AppendLogs

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerProcess.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+            throw new RunnerException("Unable to get code server logs." + e.getMessage());
+        }
+    }
+
+    @Override
+    public void onEvent(ProjectEvent event) {
+        update(event, extensionSourcesPath, projectApiBaseUrl, executor);
+    }
+
+    private static void update(final ProjectEvent event, final Path projectMirrorPath, final String projectApiBaseUrl,
```
Comment (skabashnyuk) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerProcess.java]: why this method is static?

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/WindowsTomcatServer.java (hunk 1 above)
Comment (AndrienkoAleksandr) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/WindowsTomcatServer.java]: Fixed

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServer.java (hunk 2 above)
Comment (AndrienkoAleksandr) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServer.java]: Fixed

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerImpl.java (hunk 3 above)
Comment (AndrienkoAleksandr) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerImpl.java]: Fixed

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerImpl.java (hunk 4 above)
Comment (AndrienkoAleksandr) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerImpl.java]: Fixed

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerProcess.java (hunk 5 above)
Comment (AndrienkoAleksandr) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerProcess.java]: Fixed

File: plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerProcess.java (hunk 6 above)
Comment (AndrienkoAleksandr) [inline plugin-sdk/che-plugin-sdk-runner/src/main/java/org/eclipse/che/runner/sdk/CodeServerProcess.java]: Fixed

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**